cpp/oneapi/dal/algo/pca/test/online.cpp

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/pca/test/fixture.hpp"
+
+namespace oneapi::dal::pca::test {
+
+namespace te = dal::test::engine;
+namespace la = te::linalg;
+namespace pca = oneapi::dal::pca;
+using pca_types_cov = COMBINE_TYPES((float, double), (pca::method::cov));
+using pca_types_svd = COMBINE_TYPES((float, double), (pca::method::svd));
+
+template <typename TestType>
+class pca_online_test : public pca_test<TestType, pca_online_test<TestType>> {};
+
```

Comment:
will any tests be ran on devices that are not float64 friendly? currently our CI validates on hardware that does not support float64, so all of these will be skipped

Reply:
Fixed

Reply:
And yes, it will be skipped if device doesnt support double, and the test type is double, so, it will be launched only with float

Reply:
I didn't necessarily mean to remove all checks (looks like its failing now) but instead just ensuring we are having some tests run on devices that aren't float64 friendly.